libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/utils/RouteUtils.java

```diff
@@ -178,22 +180,20 @@ public boolean isValidRouteProfile(String routeProfile) {
   @Nullable
   public BannerInstructions findCurrentBannerInstructions(LegStep currentStep, double stepDistanceRemaining) {
     if (isValidStep(currentStep) && hasInstructions(currentStep.bannerInstructions())) {
-      int roundedDistanceRemaining = (int) stepDistanceRemaining;
       List<BannerInstructions> instructions = new ArrayList<>(currentStep.bannerInstructions());
-      Iterator<BannerInstructions> instructionsIterator = instructions.iterator();
-      while (instructionsIterator.hasNext()) {
-        BannerInstructions instruction = instructionsIterator.next();
-        double distanceAlongGeometry = instruction.distanceAlongGeometry();
-        if (distanceAlongGeometry < roundedDistanceRemaining) {
-          instructionsIterator.remove();
+      Collections.sort(instructions, new Comparator<BannerInstructions>() {
+        @Override
+        public int compare(BannerInstructions instructions1, BannerInstructions instructions2) {
```

Comment:
Can we update the naming of `instructions1` and `instructions2` to something like `instructions` and `nextInstructions`?  cc @Guardiola31337

Reply:
Yes we can